apps/dg/views/component_view.js

```diff
@@ -658,6 +717,81 @@ DG.ComponentView = SC.View.extend(
+                tInspectorHeight = tButtons[0].parentView.layout.height;
+              } else {
+                var BUTTONHORIZONTALSPACE = 50,
+                    BUTTONVERTICALSPACE = 43;
+                tInspectorWidth = BUTTONHORIZONTALSPACE;
+                tInspectorHeight = BUTTONVERTICALSPACE * tButtons.length;
+              }
+            }
+            return {width : tInspectorWidth, height : tInspectorHeight};
+        },
+        enforceViewBounds : function () {
+          var tTitleBar = this.getPath('containerView.titlebar'),
+              type = this.getPath('controller.model.type'),
+              tInspectorDims = this.getInspectorDimensions(),
+              tLayout = this.getPath('layout'),
+              tIsVisible = this.getPath('isVisible'),
+              tContainerWidth = tTitleBar.getContainerWidth(),
+              tContainerHeight = tTitleBar.getContainerHeight(),
+              tScaleFactor = DG.currDocumentController().get('scaleFactor'),
+              tScaleBoundsX = DG.currDocumentController().get('scaleBoundsX'),
+              tScaleBoundsY = DG.currDocumentController().get('scaleBoundsY'),
+              tMinWidth = this.get('contentMinWidth') || kMinSize,
+              tMinHeight = this.get('contentMinWidth') || kMinSize;
+              if (tIsVisible &&
+                  tScaleBoundsX < (tLayout.left + tLayout.width + tInspectorDims.width)) {
+                var tNewBoundsX = tLayout.left + tLayout.width + tInspectorDims.width;
+                DG.currDocumentController().set('scaleBoundsX', tNewBoundsX);
+              }
+              if (tIsVisible &&
+                  (tScaleBoundsY < (tLayout.top + tLayout.height) ||
+                  tScaleBoundsY < (tLayout.top + tInspectorDims.height))) {
+                var tHeight = (tLayout.height > tInspectorDims.height) ? tLayout.height : tInspectorDims.height;
+                var tNewBoundsY = tLayout.top + tHeight;
+                DG.currDocumentController().set('scaleBoundsY', tNewBoundsY);
+              }
+              if (type === DG.Calculator) {
+                tMinWidth = tLayout.width;
+                tMinHeight = tLayout.height;
+              }
+          var tNewWidth = Math.max(tMinWidth,
+              DG.ViewUtilities.floorToGrid(tLayout.widthOrig * tScaleFactor)),
+              tNewHeight = Math.max(tMinHeight,
+              DG.ViewUtilities.floorToGrid(tLayout.heightOrig * tScaleFactor)),
+              tNewLeft = DG.ViewUtilities.floorToGrid(tLayout.leftOrig * tScaleFactor),
+              tNewTop = DG.ViewUtilities.floorToGrid(tLayout.topOrig * tScaleFactor);
+          if ((tNewLeft + tNewWidth + tInspectorDims.width) > tContainerWidth) {
+            tNewLeft = Math.max(0,  tContainerWidth - tNewWidth - tInspectorDims.width);
+          }
+          if (tNewTop + tNewHeight > tContainerHeight) {
+            tNewTop = Math.max(0, tContainerHeight - tNewHeight);
+          }
+          this.adjust('width', tNewWidth);
```

Comment:
These 4 statements can be combined: `this.adjust({width: tNewWidth, height: tNewHeight, etc.})`

Reply:
Sounds good.  I'll change.